### PR TITLE
Draft: add converter to update batch processor

### PR DESCRIPTION
**Pull request by asreehari-splunk**
when sapm exporter enables token_access_passthrough use this converter to add include_metadata with the right header key

**Description:** <Describe what has changed.>

**Link to Splunk idea:** <Link to Splunk idea, see https://ideas.splunk.com>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: internal/settings/settings_test.go [hunk 1]
```diff
@@ -337,12 +337,18 @@ func TestSetDefaultEnvVarsSetsInterfaceFromConfigOption(t *testing.T) {
...
+			"another-gate",
+		},
+		{
+			"--feature-gates", "some-gate", "--feature-gates", "+telemetry.useOtelForInternalMetrics",
+			"--feature-gates", "another-gate",
+		},
+		{
+			"--feature-gates", "some-gate", "--feature-gates", "-telemetry.useOtelForInternalMetrics",
+			"--feature-gates", "another-gate",
+		},
```
Comment (dmitryax) [inline internal/settings/settings_test.go:351]: Please avoid unrelated changes

File: internal/configconverter/update_batchproc_on_token_passthrough.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+	"go.opentelemetry.io/collector/confmap"
+)
+
+func UpdateBatchProcOnTokenPassthrough(_ context.Context, in *confmap.Conf) error {
+	if in == nil {
+		return nil
+	}
+
+	out := map[string]interface{}{}
```
Comment (dmitryax) [inline internal/configconverter/update_batchproc_on_token_passthrough.go:43]: nit: same for other places
```suggestion
	out := map[string]any{}
```

**Review by samiura — COMMENTED** [review]
:LGTM

**Review by dmitryax — COMMENTED** [review]
Please update the default config file `agent_config.yaml` as well

File: internal/configconverter/update_batchproc_on_token_passthrough.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+	// Check for sapm receiver with access_token_passthrough
+	if in.IsSet("receivers::sapm::access_token_passthrough") && in.Get("receivers::sapm::access_token_passthrough").(bool) {
```
Comment (pjanotti) [inline internal/configconverter/update_batchproc_on_token_passthrough.go:46]: I'm assuming that if there is a typo on the value of `access_token_passthrough`, e.g.: `truue`, this code is never reached, is that correct?

Can we guarantee that the type assertion never fails here?

File: internal/configconverter/update_batchproc_on_token_passthrough_test.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
+// Copyright The OpenTelemetry Authors
```
Comment (pjanotti) [inline internal/configconverter/update_batchproc_on_token_passthrough_test.go:1]: Double license header - I wish addlicense caught those...

**Review by pjanotti — COMMENTED** [review]
nit & personal preference: instead of `_w_` and `_wo_` on the test files I would rather use `_with_` and `_no_`, up to you.

File: internal/configconverter/update_batchproc_on_token_passthrough.go (hunk 3 above)
Comment (asreehari-splunk) [reply internal/configconverter/update_batchproc_on_token_passthrough.go:46]: thats right. it won't execute this code. let me think about that

File: internal/configconverter/update_batchproc_on_token_passthrough.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+func UpdateBatchProcOnTokenPassthrough(_ context.Context, in *confmap.Conf) error {
+	if in == nil {
+		return nil
+	}
+
+	out := map[string]any{}
+
+	// Check for sapm receiver with access_token_passthrough
+	if in.IsSet("receivers::sapm::access_token_passthrough") {
```
Comment (dmitryax) [inline internal/configconverter/update_batchproc_on_token_passthrough.go:32]:
```suggestion
	if in.IsSet("receivers::sapm::include_metadata") {
```

File: internal/configconverter/update_batchproc_on_token_passthrough.go [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+func UpdateBatchProcOnTokenPassthrough(_ context.Context, in *confmap.Conf) error {
+	if in == nil {
+		return nil
+	}
+
+	out := map[string]any{}
+
+	// Check for sapm receiver with include_metadata set to true
+	if in.IsSet("receivers::sapm::include_metadata") {
+		if _, ok := in.Get("receivers::sapm::include_metadata").(bool); ok {
```
Comment (dmitryax) [inline internal/configconverter/update_batchproc_on_token_passthrough.go:33]: `ok` represents whether the value casts to booled. So it's going to be true even for `include_metadata: true`. That wasn't your intention, right? Please fix and add a test case

Comment (asreehari-splunk) [reply internal/configconverter/update_batchproc_on_token_passthrough.go:33]: makes sense. I'll update it

**Comment by asreehari-splunk** [comment]
closing
